### ISSUE TITLE
"The following SSH command responded with a non-zero exit status." at batch_action.rb

### ISSUE THREAD

**Issue by MiquelAdell** [Varying-Vagrant-Vagrants/VVV#2244]
## What was The Command Used To Provision

```shell
vagrant up --provision --debug
```

## What Kind of VVV Provision Was This

This was an attempt to upgrade to a new version of VVV

## Logs/What Broke

```yaml
sites:
  // some sites that look like these
  sitename:
    repo: https://github.com/Varying-Vagrant-Vagrants/custom-site-template.git
    hosts:
      - sitename.test
    skip_provisioning: false
    custom:
      delete_default_plugins: true

  test:
    hosts:
      - test.test
    skip_provisioning: true

  wordpress-default:
    hosts:
      - local.wordpress
    repo: "https://github.com/Varying-Vagrant-Vagrants/vvv-wordpress-default.git"
    skip_provisioning: true

  wordpress-develop:
    hosts:
      - src.wordpress-develop
      - build.wordpress-develop
    repo: "https://github.com/Varying-Vagrant-Vagrants/vvv-wordpress-develop.git"
    skip_provisioning: true

  vvv:
    hosts:
      - vvv.test
    skip_provisioning: true

utilities:
  core:
    - memcached-admin
    - opcache-status
    - phpmyadmin
    - webgrind
    - tls-ca

vm_config:
  cores: 1
  memory: 2048

```

## Steps to Reproduce (for bugs)

1. I opened **virtualbox** it said I had to update. I did, to **6.1**
2. After trying to vagrant up something failed so I pulled from dev 
3. vagrant-goodhosts  failed but I've seen at the github page that there was some command to run
4. ran ```vagrant plugin install vagrant-goodhosts --local```
5. ```vagrant up``` or ```vagrant up --provision``` keep failing

Full log from here https://gist.github.com/MiquelAdell/0856c3594dee6628d42c21719ac928e4

The last lines of it are:
```shell
/opt/vagrant/embedded/gems/2.2.9/gems/vagrant-2.2.9/lib/vagrant/action/runner.rb:89:in `run'
/opt/vagrant/embedded/gems/2.2.9/gems/vagrant-2.2.9/lib/vagrant/machine.rb:240:in `action_raw'
/opt/vagrant/embedded/gems/2.2.9/gems/vagrant-2.2.9/lib/vagrant/machine.rb:212:in `block in action'
/opt/vagrant/embedded/gems/2.2.9/gems/vagrant-2.2.9/lib/vagrant/environment.rb:613:in `lock'
/opt/vagrant/embedded/gems/2.2.9/gems/vagrant-2.2.9/lib/vagrant/machine.rb:198:in `call'
/opt/vagrant/embedded/gems/2.2.9/gems/vagrant-2.2.9/lib/vagrant/machine.rb:198:in `action'
/opt/vagrant/embedded/gems/2.2.9/gems/vagrant-2.2.9/lib/vagrant/batch_action.rb:86:in `block (2 levels) in run'
 INFO interface: error: The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

echo; printf $SSH_AUTH_SOCK

Stdout from the command:

Stderr from the command:

printf: usage: printf [-v var] format [arguments]
```

## Your Environment

Varying Vagrant Vagrants: v3.5.0-git::develop
Vagrant: v2.2.9, 
VirtualBox: v6.1.14r140239

Platform: platform-darwin19.2.0 shell:/bin/zsh vagrant-hostmanager vagrant-hostsupdater vagrant-goodhosts vagrant-vbguest shared_db_folder_disabled, VVV Path: "/Users/miqueladell/code/VVV"

Thanks a lot!

**Comment by tomjn** [comment]
Do you have the full output? Your snippet cuts off all the important
information so there are no clues as to what happened in your issue. If I
can see the full output without the —debug flag?

**Comment by MiquelAdell** [comment]
Hi,

Thanks a lot for the quick reply.

If I run ```vagrant up --provision --debug``` it does cut at the line I pasted on the ghist.

If I run it again it gives a different error.

I'm pasting that here:
https://gist.github.com/MiquelAdell/ea80a6232793b0dfd14fd0d52d16cfa8

That may be because the first run of vagrant up does leave the vm machine up.

In fact, just after the first log exists with the "empty" error I can vagrant ssh into the machine (but the websites do not work).

If I after the first run I go to VM irtualbox I see the machine as up.

If I power off the machine and try again I get the error from the gist in the first post again

Thanks again!

**Comment by tomjn** [comment]
Thanks but can you do it without —debug? That flat adds so much extra
information it’s impossible to see anything useful

**Comment by MiquelAdell** [comment]
Sorry I didn't realise you asked for the full otupt without the debug flag, here it is:

first run:
https://gist.github.com/MiquelAdell/f6fdaf50eb7ff734bf784c9e07f259a1

second run:
https://gist.github.com/MiquelAdell/e68c373ef6d1f765bd5202c1e1f2aa00

**Comment by tomjn** [comment]
The first one implies a missing SSH Agent somewhere, the easiest first
debugging step I can think of is to reboot your computer and try again

**Comment by MiquelAdell** [comment]
I already did that. Same outputs.

I'll paste what I thought of, I hope it is useful in some way but I'm shooting a bit in the dark, sorry:

It's a Mac Os 10.15.6 

As far as I know nothing changed on the OS or machine side. No VPN or tunnels or anything like that at the moment

I can ping the vvv.test site
```shell
→ ping vvv.test
PING miqueladellca.test, (192.168.50.4): 56 data bytes
64 bytes from 192.168.50.4: icmp_seq=0 ttl=64 time=0.316 ms
64 bytes from 192.168.50.4: icmp_seq=1 ttl=64 time=0.325 ms
^C
--- miqueladellca.test, ping statistics ---
2 packets transmitted, 2 packets received, 0.0% packet loss
round-trip min/avg/max/stddev = 0.316/0.321/0.325/0.005 ms
```

and ```vagrant ssh``` into it
```shell
→ vagrant ssh
    ✧ Hello Again! ✧         ✧            ✧
✧   ▄▀▀▀▄▄▄▄▄▄▄▀▀▀▄   ✧  __ __ __ __
   ✧█▒▒░░░░░░░░░▒▒█      \ V\ V\ V /
 ▄▄  █░░█░░░░░█░░█ ✧      \_/\_/\_/
█░░█  █░░░▀█▀░░░█     ✧              ✧
█░░█  ▀▄░░░░░░░▄▀ ✧      Have a nice day :)


You have new mail.
Last login: Mon Sep 21 10:21:55 2020 from 192.168.50.1
vagrant@vvv:~$
```
but it does refuse my :80 connections

```shell
→ curl -i vvv.test
curl: (7) Failed to connect to vvv.test port 80: Connection refused
```

I guess the nginx server did not start. Inside the vagrant machine I get:
```shell 
vagrant@vvv:~$ sudo systemctl status nginx
● nginx.service - nginx - high performance web server
   Loaded: loaded (/lib/systemd/system/nginx.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Tue 2020-09-22 08:14:04 UTC; 9min ago
     Docs: http://nginx.org/en/docs/
  Process: 1022 ExecStart=/usr/sbin/nginx -c /etc/nginx/nginx.conf (code=exited, status=1/FAILURE)

Sep 22 08:14:04 vvv systemd[1]: Starting nginx - high performance web server...
Sep 22 08:14:04 vvv nginx[1022]: nginx: [alert] could not open error log file: open() "/var/log/nginx/error.log" failed (30: Read-only file system)
Sep 22 08:14:04 vvv nginx[1022]: 2020/09/22 08:14:04 [warn] 1022#1022: duplicate MIME type "text/html" in /etc/nginx/nginx.conf:84
Sep 22 08:14:04 vvv nginx[1022]: 2020/09/22 08:14:04 [emerg] 1022#1022: cannot load certificate "/etc/nginx/server-2.1.0.crt": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/etc/nginx/ser
Sep 22 08:14:04 vvv systemd[1]: nginx.service: Control process exited, code=exited status=1
Sep 22 08:14:04 vvv systemd[1]: nginx.service: Failed with result 'exit-code'.
Sep 22 08:14:04 vvv systemd[1]: Failed to start nginx - high performance web server.
```

**Comment by tomjn** [comment]
Do you have the bindfs plugin installed? If so does removing/disabling it then reprovisioning resolve the issue? What does `sudo nginx -t` say inside the VM?

**Comment by tomjn** [comment]
Also, if you can try and remove other vagrant plugins you don't use. For VVV you don't need hostsmanager or hostsupdater if goodhosts is installed

**Comment by MiquelAdell** [comment]
Sorry it took me a bit.

I do have the bindfs plugin installed. 

I have the following:
```shell
→ vagrant plugin list
vagrant-bindfs (1.1.8, global)
vagrant-exec (0.5.3, global)
  - Version Constraint: > 0
vagrant-goodhosts (1.0.0.pre.beta4, global)
  - Version Constraint: 1.0.0.pre.beta4
vagrant-hostsupdater (1.1.1.160, global)
  - Version Constraint: > 0
vagrant-vbguest (0.24.0, global)
  - Version Constraint: > 0
```

I just removed vagrant-hostmanager. (no changes with the issue)

Not sure which of thoose are needed. I'm a bit of a copypaster in this area, sorry.

The output of nginx -t is:

```shell
vagrant@vvv:~$ sudo nginx -t
nginx: [alert] could not open error log file: open() "/var/log/nginx/error.log" failed (30: Read-only file system)
2020/09/22 09:42:30 [warn] 1361#1361: duplicate MIME type "text/html" in /etc/nginx/nginx.conf:84
2020/09/22 09:42:30 [emerg] 1361#1361: cannot load certificate "/etc/nginx/server-2.1.0.crt": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/etc/nginx/server-2.1.0.crt','r') error:2006D080:BIO routines:BIO_new_file:no such file)
nginx: configuration file /etc/nginx/nginx.conf test failed
```

I tried running ```vagrant plugin expunge --reinstall``` and also removing the ```.vagrant/plugins``` folder and running ```vagrant plugin update```. No changes.


I also tried completely removing virtualbox 6.1 and reinstalling 6.1 because this sounded familiar:
https://github.com/hashicorp/vagrant/issues/11058

But I get the same.

I can downgrade virtualbox to 6.0 if that would help the debuging.

Thanks!

**Comment by Mte90** [comment]
I have the same version of vagrant and virtualbox and no issues.
Are you on linux or osx? seems that the issues are mainly file permissions.
Did you was using with sudo VVV?

**Comment by MiquelAdell** [comment]
osx 10.15.6. 
I'm running vv as a regular user.
The permissions shouldn't have changed since yesterday. 

I just tested downgrading to  VirtualBox: v6.0.24r139119 and got the same results.

I'm noticing two things I did not notice before.

When I do ```vagrant up```
I get:

```shell
→ vagrant up
__ __ __ __
\ V\ V\ V / Varying Vagrant Vagrants
 \_/\_/\_/  v3.5.0-git::develop

Platform: platform-darwin19.2.0 shell:/bin/zsh vagrant-hostsupdater vagrant-goodhosts vagrant-vbguest shared_db_folder_disabled, VVV Path: "/Users/miqueladell/code/VVV"
Vagrant: v2.2.9, VirtualBox: v6.0.24r139119

Docs:       https://varyingvagrantvagrants.org/
Contribute: https://github.com/varying-vagrant-vagrants/vvv
Dashboard:  http://vvv.test
```
and some lines after that this:
```shell
==> default: [vagrant-goodhosts] Checking for host entries
==> default: [vagrant-goodhosts] Warning: The vagrant-hostsupdater plugin is installed, hostsupdater always adds the VM name even if the VagrantFile says not to. This shouldn't cause issues, but if it does, uninstall hostsupdater
==> default: [vagrant-goodhosts] - found entry for: 192.168.50.4 vvv
```

I tried removing vagrant-hostsupdater with ```vagrant plugin uninstall vagrant-hostsupdater``` but I still see the same.

Also ```vagrant plugin list``` shows the following:
```shell
→ vagrant plugin list
vagrant-bindfs (1.1.8, global)
vagrant-exec (0.5.3, global)
  - Version Constraint: > 0
vagrant-goodhosts (1.0.0.pre.beta4, global)
  - Version Constraint: 1.0.0.pre.beta4
vagrant-vbguest (0.24.0, global)
  - Version Constraint: > 0
```

Where is that vagrant-hostsupdater coming from? How can I remove it?

I don't think this is the issue but I'm trying to remove as many moving parts as possible.

Thanks!

**Comment by Mte90** [comment]
Well the error https://github.com/Varying-Vagrant-Vagrants/VVV/issues/2244#issuecomment-696622731 mention the fact that doesn't have permissions to write files so the real issues it is here for me.
Did you try a `vagrant destroy` and `vagrant up`?

**Comment by MiquelAdell** [comment]
It did work!
Thanks!

I'll try re-updating to virtualbox 6.1 and comment again and close if it works.

Thanks again, I really needed to solve that

**Comment by Mte90** [comment]
You are welcome I think that in the past you used sudo or there was some misconfiguration of permission inside the VM so a destroy will fix :-)

**Comment by MiquelAdell** [comment]
**TL;DR:** in case you are looking for the answer to the problem in the closing post, for me, the solution was here:
https://github.com/Varying-Vagrant-Vagrants/VVV/issues/2244#issuecomment-696663920


I just updated to VirtualBox: v6.1.14r140239 and everything still works.

Thanks @Mte90 and @tomjn for your thoroughness and quick responses.
